radar-auth/src/main/java/org/radarcns/auth/authentication/TokenValidator.java

```diff
@@ -96,6 +119,8 @@ private void loadPublicKey() throws TokenValidationException {
         lastFetch = Instant.now();
```

Comment:
The purpose of this lastFetch is to prevent a client from being able to make the application that uses this library to overwhelm the token endpoint. A malicious user might send requests with bad tokens indefinitely, which would cause us to continuously fetch a new public key.
As long as clients supply correctly signed tokens, the key is never refetched.
I updated the comments on the `validateAccessToken()` method to better explain the behavior.

Reply:
Also moving the check on lastFetch to `loadPublicKey()`, this will prevent multiple threads that are blocked on `loadPublicKey()` to all fetch the public key once it's their turn to enter the method.